test/Spaces/terrain_warp.jl

```diff
@@ -128,9 +151,10 @@ function warpedspace_2D(
     warp_fn = warp_sin_2d,
     test_smoothing = false,
     adaption = Hypsography.LinearAdaption,
-)
+    device,
+) where {FT}
     vert_face_space, hspace =
-        generate_base_spaces(xlim, zlim, helem, velem, npoly, ndims = 2)
+        generate_base_spaces_2d(xlim, zlim, helem, velem, npoly; device)
```

Comment:
Follows on from `device` assumption for `SpectralElementSpace2D`; as I recall there is no GPU support for these.

Reply:
No, but that doesn’t mean we can’t hoist the call out